packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -208,62 +217,78 @@ export class EthersEth1Notifier implements IEth1Notifier {
       this.logger.verbose(`Processing ${blockDepositEvents.length} deposit events of eth1 block ${blockNumber}`);
       this.lastDepositCount = blockDepositEvents[blockDepositEvents.length - 1].index + 1;
     }
-    const shouldGetBlock = this.lastDepositCount >= this.config.params.MIN_GENESIS_ACTIVE_VALIDATOR_COUNT &&
-      this.passCheckpoint(blockNumber);
     // preGenesis: avoid calling getBlock() frequently if we are too far away from genesis time
-    // postGenesis: always call getBlock to store eth1Data
-    if (shouldGetBlock) {
+    // postGenesis: avoid calling getBlock() frequently until sync is completed
+    if (this.shouldGetBlock(blockNumber)) {
       const block = await this.getBlock(blockNumber);
       if (!block) {
         this.logger.verbose(`eth1 block ${blockNumber} not found`);
         return false;
       }
-      const beforeGenesis = this.setCheckpoint(block);
-      if (beforeGenesis) {
-        this.eth1Source && blockDepositEvents.length > 0 && this.eth1Source.push({events: blockDepositEvents});
-        return true;
-      } else {
-        // maybe no deposit events for this block, still need to push to form genesis
-        this.eth1Source && this.eth1Source.push({events: blockDepositEvents, block});
-        return await this.processEth1Data(block);
-      }
+      // set the next checkpoint
+      const shouldProcessEth1Data = this.setCheckpoint(block);
+      // maybe no deposit events for this block, still need to push to form genesis
+      if (this.eth1Source) this.eth1Source.push({events: blockDepositEvents, block});
+      if (shouldProcessEth1Data) return await this.processEth1Data(block);
     } else {
       this.eth1Source && blockDepositEvents.length > 0 && this.eth1Source.push({events: blockDepositEvents});
     }
     return true;
   }
 
   /**
-   * Before genesis: return blockNumber >= checkpoint
-   * After genesis: true
+   * Decide if we should get eth1 block or not.
+   */
+  public shouldGetBlock(blockNumber: number): boolean {
+    const passed = this.passCheckpoint(blockNumber);
+    return (this.eth1Source)?
+      this.lastDepositCount >= this.config.params.MIN_GENESIS_ACTIVE_VALIDATOR_COUNT && passed : passed;
```

Comment:
I'm not sure I get logic here?

Reply:
this is for building genesis state: we don't need to get eth1 block if we don't have enough active validator.
the code was already in place, just doing a refactor here.